synapse/lib/ast.py

```diff
@@ -101,16 +102,27 @@ def prepare(self):
         pass
 
     def hasAstClass(self, clss):
+        hasast = self.hasast.get(clss)
+        if hasast is not None:
+            return hasast
```

Comment:
Would you mind adding a test to mop up covering this line?  it's small but important :D